### ISSUE TITLE
"Error: listen EACCES: permission denied 0.0.0.0:80"

### ISSUE THREAD

**Issue by dynamicdave1515** [datech/node-red-contrib-amazon-echo#1]
Just tried your node in Node-RED v0.20.0 and got this error message from the Hub node

"Error: listen EACCES: permission denied 0.0.0.0:80"

**Comment by datech** [comment]
Ports under 1024 needs Node-Red to be started with root user or you can use docker containers, like Home Assistance is doing to run Node-Red. I will update the documntation to refrect this dowside

**Comment by Rapejim** [comment]
I am running Node-RED with a docker container and I configure the port 80 to be published port:



![image](https://user-images.githubusercontent.com/7972545/50963163-3d8c4e80-14cc-11e9-8e47-abda68153ab3.png)

But I recive the same error:
"Error: listen EACCES 0.0.0.0:80"
![image](https://user-images.githubusercontent.com/7972545/50963200-5bf24a00-14cc-11e9-9e96-3bf49d7742ec.png)

**Comment by datech** [comment]
Is process running with root user inside the docker container?

**Comment by Rapejim** [comment]
I think it's not running as root:
![image](https://user-images.githubusercontent.com/7972545/50965388-19cc0700-14d2-11e9-9e7a-1d4b2b6909d3.png)
But, I don't know how change it (inside the container).

**Comment by datech** [comment]
You may try to override the docker user with --user=root  argument on docker run command

This has to switch node-red user to root

https://docs.docker.com/engine/reference/run/#user

**Comment by Rapejim** [comment]
> You may try to override the docker user with --user=root argument on docker run command
> 
> This has to switch node-red user to root
> 
> https://docs.docker.com/engine/reference/run/#user
> 
> PS: Port forwarding of port 80 should be working too

SOLVED!! Y have changed the run user to root (in Portainer, the docker webui) and now it's "conected" without port error.
![image](https://user-images.githubusercontent.com/7972545/50967545-b4c7df80-14d8-11e9-9852-ac5f01806f12.png)
But Alexa don't discover the new device "Cargador patín"... (I have Echo Dot 3gen). 🤔

**Comment by Rapejim** [comment]
I have this info:
![image](https://user-images.githubusercontent.com/7972545/50987034-bd380e80-1508-11e9-90db-bdc82e08f5b4.png)

**Comment by datech** [comment]
Thanks. 

Ok, this is the expected XML.

That’s odd. I suppose, that both Node-Red and Echo are in the same network?

**Comment by Rapejim** [comment]
Node-Red (on a Raspi) and Echo are connected at the same wifi (my unique router).

**Comment by datech** [comment]
I managed to reproduce it. The problem is that Amazon Echo Hub could not be discovered with SSDP.
If the container is running in isolated network it multicast messages cannot be delivered to it.

In order to fix it you have to select **host** network while deploying the docker container. Here is how you can do in Portainer: 
<img width="494" alt="node-red amazon echo alexa" src="https://user-images.githubusercontent.com/535026/50993531-9686d180-1522-11e9-8a7d-3dec8c4d32d8.png">

**Comment by Rapejim** [comment]
> I managed to reproduce it. The problem is that Amazon Echo Hub could not be discovered with SSDP.
> If the container is running in isolated network, multicast messages cannot be delivered to it.
> 
> In order to fix it you have to select **host** network while deploying the docker container. Here is how you can do in Portainer:
> <img alt="node-red amazon echo alexa" width="494" src="https://user-images.githubusercontent.com/535026/50993531-9686d180-1522-11e9-8a7d-3dec8c4d32d8.png">

Yeah, this change solve de ports problem. 
But when I add new node, (for new device), I must restart NodeRED, then Alexa detect the new device. If I don't restart NodeRED, Alexa don't detect new device...
The Mysteries of Technology 🤣

**Comment by datech** [comment]
Echo Dot 2 gen has to be able to connect to any port above 1024.

PS: I don't have Dot 2 Gen to test it.

**Comment by datech** [comment]
Hello David

I will close this issue as there is a confirmation that the packages is working with Echo Dot 2 Gen.

https://discourse.nodered.org/t/amazon-echo-hub/6724/6

Fill free to open another issue if you find any bug or missing functionality.

Thanks.

**Comment by ChazzerB314** [comment]
> 
> 
> > You may try to override the docker user with --user=root argument on docker run command
> > This has to switch node-red user to root
> > https://docs.docker.com/engine/reference/run/#user
> > PS: Port forwarding of port 80 should be working too
> 
> SOLVED!! Y have changed the run user to root (in Portainer, the docker webui) and now it's "conected" without port error.
> ![image](https://user-images.githubusercontent.com/7972545/50967545-b4c7df80-14d8-11e9-9852-ac5f01806f12.png)


I'm having this same issue, but I went to that link I have no idea what it's saying. Can you help me dumb it down a bit? I'm still very new to all of this.

**Comment by cbrandlehner** [comment]
My node-red environment is running non-root and I did not want to change this.

To be able to bind to port 80 you can also use this command to allow non-root processes to bind to port 80.

`sysctl -w net.ipv4.ip_unprivileged_port_start=80`
